Updates (#102)
bcgov/bcdg: docs/developer-community/ms-teams-contractors-access.md

# Microsoft Teams access for contractors

## Overview

> Access to Microsoft Teams for contractors depends on whether the other features of an IDIR account are required for their role. Note that contracted developers who want to use GitHub organizations [bcgov](https://github.com/bcgov) / [bcgov-c](https://github.com/bcgov-c) will require a P2 or higher IDIR. For more information visit the [Github technical transition guide](../use-github-in-bcgov/github-technical-transition%20guide.md)

Author: ```suggestion
Access to Microsoft Teams for contractors depends on whether the other features of an IDIR account are required for their role. Note that contracted developers who want to use GitHub organizations [bcgov](https://github.com/bcgov) / [bcgov-c](https://github.com/bcgov-c) will require a P2 or higher IDIR. For more information visit the [Github transition guide](../use-github-in-bcgov/github-transition-guide.md)
```

> 

Reviewer: Wonder about linking to the `github-transition-guide.md` page instead as it offers the details on the IDIR levels plus how to join bcgov and bcgov-c orgs

I know the callout below links to `github-transition-guide.md` page. But the  `github-technical-transition%20guide.md` seems like a very point in time guide of when we did the transition to SSO

## Contractors with IDIR accounts

!!! info "Identify your IDIR account"
    To identify the IDIR account type visit [IDIR and technical guide for SSO](../use-github-in-bcgov/github-transition-guide.md#compatible-idirs-for-contractors).

### Contractors who already have IDIR (with an E5 License)

Contractors with E5 licences get the same access as government staff. Use the same process to add them. By default, they can access any site that uses “Everyone except Guests” permissions.

### Contractors who already have IDIR (without an E5 License)

Contractors who already have an IDIR account but do not have an E5 licence need to complete these steps:

**Step 1: Associate an external email address** 

Ask an authorized requestor to submit an **Associate an IDIR with an external email address** request through My Service Centre.

**Step 2: Request hybrid cloud conversion**

> Email CSCHelp@gov.bc.ca and ask convert the IDIR account to a [Hybrid Cloud account](https://ociomysc.service-now.com/now/knowledge-center/kb_view/kb_knowledge/83745ccb33d376107f71fe282e5c7b29).

Reviewer: ```suggestion
Email CSCHelp@gov.bc.ca and ask to convert the IDIR account to a [Hybrid Cloud account](https://ociomysc.service-now.com/now/knowledge-center/kb_view/kb_knowledge/83745ccb33d376107f71fe282e5c7b29).
```

Submit an onboarding request through **My Service Centre** for an IDIR account and include the contractor's external email address in the request details. 

**Step 2: Request hybrid cloud conversion**

Email CSCHelp@gov.bc.ca and ask convert the IDIR account to a [Hybrid Cloud account](https://ociomysc.service-now.com/now/knowledge-center/kb_view/kb_knowledge/83745ccb33d376107f71fe282e5c7b29).

This conversion gives the contractor one identity to access B.C. government resources. It also lets them use their existing organization login to access Microsoft Teams and SharePoint.

## Existing contractors without IDIR accounts 

Contractors without an IDIR account can join Microsoft Teams as **guest users**. They do not get access to IDIR features. When their contract ends, the service team that gave them initial access must remove their access manually.

* Teams owners and administrators can manage guest users the same way they manage internal users
* Guest users stay active after the contract end date because the system does not remove them automatically
* IDIR accounts deactivate automatically when they expire. However, all guest accounts need manual removal

> 

Reviewer: Do we want to call out more (maybe as part of the warning box or something) about the onus is on the Team Owners to keep their MS Teams chat/channel membership up to date.

Offboarding seems pretty spotty, with some teams on top of it, and others not.

For help with external guest accounts, email MFA@gov.bc.ca.

!!! warning 
    "Guest access in Microsoft Teams is currently available at no cost. This process may change in the future."

## Easy flowchart guide

```mermaid
flowchart TD

    A{Does the contractors have an IDIR?}

    A -->|<b>Yes</b>| B{Is it an E5 license<br/>or P2 license?}

>     A -->|<b>No</b>| C[They will be added as a <b>Guest user</b> in MS Teams]

Reviewer: With the diagram it seems like the option is to just add them as a guest rather than consider getting them a license, if they don't have a license already.

 Should there be a step to consider getting a license before defaulting to adding them as guest. As guest users aren't automatically cleaned up.

Author: I was curious about this step, is it frown upon to NOT get an IDIR overall? Should we default them to get one? If so, do they need P2 or E5 licences? @MonicaG

Reviewer: I don't know. I think that is my question too. 

Maybe something to ask Charles?

Or maybe we don't default either way and add in a step titled something like:

"Decide if contractor will have IDIR"

Reviewer: What was the decision on this? Leave the diagram as is, add a decision node, something else?

Author: Yes, asking Charles now about it, I just wanted to fix previous comments and typos I found.

    C --> D[Request Teams owners or administrators to add them as Guest users]
    D --> E["Owners are identified in Manage team<br/>(More options → Manage team → Add member)<br/>or send a request to the owners"]

    B -->|<b>E5 license</b>| F[Contractor has an E5 license]
    F --> G([They are all set<br/>No action needed])

    B -->|<b>P2 license</b>| H[Contractor has a P2 license]
    H --> I[Convert the IDIR to a Hybrid Cloud Account]
    I --> J[Send a request to CSCHelp@gov.bc.ca]

    K["Need help identifying the IDIR account type?"]

>     B -.-> K["Need help identifying the IDIR account type?<br/><a href='../use-github-in-bcgov/github-transition-guide#compatible-idirs-for-contractors'>Open IDIR technical guide for SSO</a>"]

Reviewer: This link 404s

>     B -.-> K["Need help identifying the IDIR account type?<br/><a href='../use-github-in-bcgov/github-transition-guide#compatible-idirs-for-contractors'>Open IDIR technical guide for SSO</a>"]

Author: ```suggestion
    B -.-> K["Need help identifying the IDIR account type?<br/><a href='https://developer.gov.bc.ca/docs/default/component/bc-developer-guide/use-github-in-bcgov/github-transition-guide/'>Open IDIR technical guide for SSO</a>"]
```

Email CSCHelp@gov.bc.ca and ask convert the IDIR account to a [Hybrid Cloud account](https://ociomysc.service-now.com/now/knowledge-center/kb_view/kb_knowledge/83745ccb33d376107f71fe282e5c7b29).

This conversion gives the contractor one identity to access B.C. government resources. It also lets them use their existing organization login to access Microsoft Teams and SharePoint.

## Existing contractors without IDIR accounts 

Contractors without an IDIR account can join Microsoft Teams as **guest users**. They do not get access to IDIR features. When their contract ends, the service team that gave them initial access must remove their access manually.

* Teams owners and administrators can manage guest users the same way they manage internal users
* Guest users stay active after the contract end date because the system does not remove them automatically
* IDIR accounts deactivate automatically when they expire. However, all guest accounts need manual removal

For help with external guest accounts, email MFA@gov.bc.ca.

!!! warning "Important considerations" 
    * Guest access in Microsoft Teams is currently available at no cost. This process may change in the future

>     * All service teams need to have an offboarding procedure for Microsoft Teams channels as all guest accounts need to be removed manually. Refer to [Developer Community Microsoft Teams guide](../developer-community/developer-community-ms-teams-guide.md#how-the-developer-community-is-structured) for more guidance.

Author: ```suggestion
    * All service teams need to have an offboarding procedure for Microsoft Teams channels as all guest accounts need to be removed manually. Refer to [Developer Community Microsoft Teams guide](../developer-community/developer-community-ms-teams-guide.md#how-the-developer-community-is-structured) for more guidance
```
